fedmsg_meta_fedora_infrastructure/autocloud.py

```diff
@@ -21,14 +21,98 @@
 
 
 class AutoCloudProcessor(BaseProcessor):
+
     __name__ = "autocloud"
     __description__ = "Automated Fedora Cloud Image Testing service"
     __link__ = "https://github.com/kushaldas/autocloud"
     __docs__ = "https://github.com/kushaldas/autocloud"
     __icon__ = "https://apps.fedoraproject.org/img/icons/fedimg.png"
     __obj__ = "Cloud Image Test"
 
+
+    def _func_router(self, msg, fname, **config):
+        FNAME = 'handle_%s_autocloud_%s'
+
+        if 'compose_id' in msg['msg'] or 'compose_job_id' in msg['msg']:
+            return getattr(self, FNAME % ('new', fname))(msg, **config)
```

Comment:
Instead of using new/old you may want to do something like v1/v2, otherwise you may end having problems if you change the format once more.

Also, while I find the approach quite neat, do we loose a bit in readability?